pkg/healthcheck/healthcheck.go

```diff
@@ -273,6 +279,60 @@ func checkHealth(serverURL *url.URL, backend *BackendConfig) error {
 	return nil
 }
 
+// checkHealthGrpc returns a nil error in case it was successful and otherwise
+// a non-nil error with a meaningful description why the health check failed.
+// Dedicatted to gRPC servers implementing gRPC Health Checking Protocol v1.
+func checkHealthGrpc(serverURL *url.URL, backend *BackendConfig) error {
+	u, err := serverURL.Parse(backend.Path)
+	if err != nil {
+		return fmt.Errorf("failed to parse serverURL: %w", err)
+	}
+	grpcSrvAddr := u.Hostname() + ":" + u.Port()
+
+	opts := []grpc.DialOption{
+		grpc.WithInsecure(),
```

Comment:
You need to handle the scheme correctly with `Insecure` only with "http" scheme.

Reply:
Now grpc Insecure opt is only enabled for `http` and `h2c`